### PR TITLE
Migrate to .jenkins.yml

### PR DESCRIPTION
**Pull request by nre-ableton**
ping @AbletonDevTools/gotham-city

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -87,7 +83,7 @@ runTheBuilds.runDevToolsProject(
               // that call fails, then we should push this image to the registry.
               docker.image("abletonag/groovylint:${versionNumber}").pull()
             } catch (ignored) {
-              data['image'].push(VersionTagger.majorMinorVersion(versionNumber))
+              data['image'].push(version.majorMinorVersion(versionNumber))
```
Comment (ala-ableton) [inline Jenkinsfile]: In 0.11, the `version` singleton doesn't have a `majorMinorVersion` function. This commit needs to be squashed into https://github.com/AbletonDevTools/groovylint/pull/65/commits/46e60cac39d25eea5b085b377d6bd18f6fccc4b8 (_Migrate to one-job-per-branch structure_).

Comment (nre-ableton) [reply Jenkinsfile]: 🤔 ... hmm, unfortunately it's not so simple. If we remove the `import` statement for `VersionTagger`, we can't reference that static function anymore. So it seems we need to squash these two commits together entirely. That will make this PR a bit different than the others in terms of history, but I think we can make an exception in this case.

Comment (ala-ableton) [reply Jenkinsfile]: > it seems we need to squash these two commits together entirely

That's exactly what I said:

> This commit needs to be squashed into 46e60ca

Comment (nre-ableton) [reply Jenkinsfile]: Ah sorry, I think I misunderstood you there. Will do!

Comment (nre-ableton) [reply Jenkinsfile]: Hmm, when I tried to do this rebase, it ended up making b4bfe38 obsolete. @ala-ableton are you ok with this PR being a single atomic commit (with an extended message)?

Comment (ala-ableton) [reply Jenkinsfile]: I don't understand. https://github.com/AbletonDevTools/groovylint/commit/b4bfe38c800e1ff966e9fcf7ea3c28273ebe34b2 (_Return early for deprecated runthebuilds job types_) is fine on its own, and I think it should stay as such.

Comment (nre-ableton) [reply Jenkinsfile]: @ala-ableton Done! Sorry about the confusion. 😊

Comment (ala-ableton) [reply Jenkinsfile]: @nre-ableton No problem!

**Pull request merged**